packages/synapse-react-client/src/components/DataGrid/SynapseGrid.tsx

```diff
@@ -112,13 +112,19 @@ const SynapseGrid = forwardRef<
     const { columnNames, columnOrder, rows } = modelSnapshot
     const gridRows = rows.map(row => {
       const rowObj: { [key: string]: any } = {}
-      // Use columnOrder to determine which columnNames to use and in what order
       columnOrder.forEach((index: number) => {
         const columnName = columnNames[index]
         if (columnName) {
           rowObj[columnName] = row.data[index]
         }
       })
+      const validation = row.metadata?.rowValidation
+      rowObj.validationStatus = validation
```

Comment:
Suggest restructuring the `DataGridRow` type to nest the data so this validation info does not collide with it (see comment right above this). Or at the very least, prefix this with an underscore, e.g.
```suggestion
      rowObj._validationStatus = validation
```